packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -46,16 +46,62 @@ export enum AllocationStrategy {
   SPOT_CAPACITY_OPTIMIZED = 'SPOT_CAPACITY_OPTIMIZED',
 }
 
+/**
+ * Launch template property specification
+ */
+export interface LaunchTemplateSpecification {
+  /**
+   * The Launch template name
+   */
+  readonly launchTemplateName: string;
+  /**
+   * The launch template version to be used (optional).
+   *
+   * @default the default version of the launch template
+   */
+  readonly version?: string;
+}
+
 /**
  * Properties for defining the structure of the batch compute cluster.
  */
 export interface ComputeResources {
   /**
-   * The IAM role applied to EC2 resources in the compute environment.
+   * The allocation strategy to use for the compute resource in case not enough instances of the best
+   * fitting instance type can be allocated. This could be due to availability of the instance type in
+   * the region or Amazon EC2 service limits. If this is not specified, the default is BEST_FIT, which
+   * will use only the best fitting instance type, waiting for additional capacity if it's not available.
+   * This allocation strategy keeps costs lower but can limit scaling. If you are using Spot Fleets with
+   * BEST_FIT then the Spot Fleet IAM Role must be specified. BEST_FIT_PROGRESSIVE will select an additional
+   * instance type that is large enough to meet the requirements of the jobs in the queue, with a preference
+   * for an instance type with a lower cost. SPOT_CAPACITY_OPTIMIZED is only available for Spot Instance
+   * compute resources and will select an additional instance type that is large enough to meet the requirements
+   * of the jobs in the queue, with a preference for an instance type that is less likely to be interrupted.
    *
-   * @default - a new role will be created.
+   * @default AllocationStrategy.BEST_FIT
    */
-  readonly instanceRole?: iam.IRole;
+  readonly allocationStrategy?: AllocationStrategy;
+
+  /**
+   * An optional launch template to associate with your compute resources. To use a launch template, you must
+   * specify either the launch template ID or launch template name in the request, but not both.
+   * For more information, see Launch Template Support.
+   *
+   * @default no custom launch template will be used
+   * @link https://docs.aws.amazon.com/batch/latest/userguide/launch-templates.html
+   */
+  readonly launchTemplate?: LaunchTemplateSpecification;
+
+  /**
+   * The Amazon ECS instance profile applied to Amazon EC2 instances in a compute environment. You can specify
+   * the short name or full Amazon Resource Name (ARN) of an instance profile. For example, ecsInstanceRole or
+   * arn:aws:iam::<aws_account_id>:instance-profile/ecsInstanceRole . For more information, see Amazon ECS
+   * Instance Role in the AWS Batch User Guide.
+   *
+   * @default a new role will be created.
+   * @link https://docs.aws.amazon.com/batch/latest/userguide/instance_IAM_role.html
+   */
+  readonly instanceRole?: string;
```

Comment:
They're actually two different beasts:

The `instanceRole` is actually an Instance Profile (in this case "instance" refers to the underlying EC2 resource) associated with a Role that has the `AmazonEC2ContainerServiceforEC2Role` managed policy attached. So, the closest relation is actually to the `ComputeResources` interface, which is the abstraction that always depends on it (you cannot have a `ComputeResources` implementation without a instanceRole property - in our case the actual property is optional because the default behaviour is to create one). It's also worth noting (see below) that a default Role and Instance Profile are created at first console experience on the region (roleName: ecsInstanceRole).

The `serviceRole` on the other hand is just a regular IAM Role with the `AWSBatchServiceRole` managed policy attached to it, so the resource perform its organic actions on our behalf.

So, answering your question, I've decided to use `string` on the `instanceRole` property because it seemed like the easiest way to refer to an existing instance profile when willing to use it (usually the one that is created by default when using the console for the first time in a region, namely `ecsInstanceRole`).